asapdiscovery-data/asapdiscovery/data/tests/test_ligand_comparison.py

```diff
@@ -0,0 +1,220 @@
+    assert not other.has_defined_stereo
+
+
+def test_true_chemical_comparisons(
+    base_ligand,
+    different_name,
+    stereoisomer,
+    tautomer,
+    acid,
+    acid_stereoisomer,
+):
+    assert base_ligand == base_ligand
+
+    # this is weird but bc `data_equal` ignores the name, this is true
+    assert base_ligand == different_name
+
+    assert base_ligand.is_chemically_equal(different_name)
+    assert base_ligand.is_stereoisomer(stereoisomer)
+    assert base_ligand.is_tautomer(tautomer)
+    assert base_ligand.is_protonation_state_isomer(acid)
+
+    assert acid.is_stereoisomer(acid_stereoisomer)
+    assert stereoisomer.is_protonation_state_isomer(acid_stereoisomer)
+
+
+def test_false_chemical_comparisons(
+    base_ligand, different_name, stereoisomer, tautomer, acid, acid_stereoisomer, other
+):
+    for query in [stereoisomer, tautomer, acid]:
+        assert not base_ligand.is_chemically_equal(query)
+    for query in [different_name, acid, other]:
+        assert not base_ligand.is_stereoisomer(query)
+    for query in [different_name, stereoisomer, acid, other]:
+        assert not base_ligand.is_tautomer(query)
+    for query in [different_name, stereoisomer, tautomer, other]:
+        assert not base_ligand.is_protonation_state_isomer(query)
+
+    assert not stereoisomer.is_tautomer(tautomer)
+    assert not stereoisomer.is_stereoisomer(tautomer)
+    assert not stereoisomer.is_protonation_state_isomer(acid)
+
+    assert not tautomer.is_tautomer(acid)
+    assert not tautomer.is_stereoisomer(acid)
+    assert not tautomer.is_protonation_state_isomer(acid)
+
+    assert not tautomer.is_tautomer(acid_stereoisomer)
+    assert not tautomer.is_stereoisomer(acid_stereoisomer)
+    assert not tautomer.is_protonation_state_isomer(acid_stereoisomer)
+
+    assert not acid.is_tautomer(acid_stereoisomer)
+
+
+def test_base_chemical_relationships(
+    base_ligand, different_name, stereoisomer, tautomer, acid, acid_stereoisomer, other
+):
+    assert (
+        base_ligand.get_chemical_relationship(different_name)
+        == ChemicalRelationship.IDENTICAL
+    )
+    assert (
+        base_ligand.get_chemical_relationship(stereoisomer)
+        == ChemicalRelationship.STEREOISOMER
+    )
+    assert (
+        base_ligand.get_chemical_relationship(tautomer) == ChemicalRelationship.TAUTOMER
+    )
+    assert (
+        base_ligand.get_chemical_relationship(acid)
+        == ChemicalRelationship.PROTONATION_STATE_ISOMER
+    )
+
+    assert base_ligand.get_chemical_relationship(other) == ChemicalRelationship.DISTINCT
+
+    assert (
+        acid.get_chemical_relationship(acid_stereoisomer)
+        == ChemicalRelationship.STEREOISOMER
+    )
+
+    assert (
+        stereoisomer.get_chemical_relationship(acid_stereoisomer)
+        == ChemicalRelationship.PROTONATION_STATE_ISOMER
+    )
+
+
+def test_weird_chemical_relationships(stereoisomer, tautomer, acid, acid_stereoisomer):
```

Comment:
Hahahaha, cracked me up.

Reply:
haha i gave up trying to come up with a more intelligent name